tokenizers/src/models/unigram/model.rs

```diff
@@ -407,14 +433,22 @@ impl Model for Unigram {
         let mut offset = 0;
         let mut tokens = Vec::with_capacity(str_tokens.len());
         for string in str_tokens {
-            let id: u32 = match self.token_to_ids.get(&string) {
-                Some(id) => *id,
-                None => self.unk_id.ok_or(UnigramError::MissingUnkId)? as u32,
+            let ids = if self.token_to_ids.contains_key(&string) {
+                vec![*self.token_to_ids.get(&string).unwrap()]
+            } else if self.byte_fallback {
+                string
+                    .bytes()
+                    .map(|b| self.token_to_id(&byte_to_piece(b)).unwrap())
+                    .collect()
+            } else {
+                vec![self.unk_id.ok_or(UnigramError::MissingUnkId)? as u32]
             };
-            let len = string.len();
-            let offsets = (offset, offset + len);
+            let len = string.len() - ids.len() + 1;
+            for id in ids {
+                let offsets = (offset, offset + len);
+                tokens.push(Token::new(id, self.id_to_token(id).unwrap(), offsets));
+            }
```

Comment:
Remove every `unwrap` and every `vec`.

There is 1 `collect` tolerated ( I think it's done that way in BPE) and it's *only* to check that ALL bytes have a token id (you're allowed to use a single `vec` or `collect` in that branch, not in the others.

:)

Reply:
I understand the `unwrap` part since it implicitly implies that potential errors are not being handled.

Can you share the reasoning regarding `vec`? is it because since this focuses on the bytefallback pieces, which should be known during compile time (256 of them) and therefore should be addressed with arrays?